### ISSUE TITLE
Deprecate the subscription related `jsonrpc` endpoints,e g. `/subscribe`

### ISSUE THREAD

**Issue by andynog** [cometbft/cometbft#442]
I believe that the RPC endpoints below related to event subscription might be removed and/or deprecated from the `jsonrpc` interface. There are three RPC endpoints related to event subscription:

```
http://localhost:26657/subscribe
http://localhost:26657/unsubscribe
http://localhost:26657/unsubscribe_all
```

They are specified here https://github.com/cometbft/cometbft/blob/06eeaaa6b6586ea7546ab3522b7c5a0d1831d523/rpc/core/routes.go#L13

Currently if I try to access `http://localhost:26657/subscribe`, I get an error

```
{"jsonrpc":"2.0","id":-1,"error":{"code":-32601,"message":"Method not found"}}
```

As far as I know these don't work with the `http` protocol. The right way to subscribe to events is using the `/websockets` and using the `ws://` protocol, e.g the endpoint `ws://127.0.0.1:26657/websocket` and sending a message to subscribe for `NewBlock `events:
```
{
    "jsonrpc": "2.0",
    "method": "subscribe",
    "id": 0,
    "params": {
        "query": "tm.event='NewBlock'"
    }
}
```

as described in the [documentation](https://docs.cometbft.com/v0.34/core/subscription)

if these endpoints are deprecated we will also need to ensure the [RPC OpenAPI documentation](https://github.com/cometbft/cometbft/blob/main/rpc/openapi/openapi.yaml) is up to date.

I believe that the relayers or any other operato don't use these endpoints since they doesn't seem to work, but it's also a good idea to double check with the Hermes team for example to ensure nothing breaks by removing the endpoints.

**Comment by cason** [comment]
Yes, I can confirm that subscription methods only work with the web socket protocol, using the `ws://` endpoint.

**Comment by andynog** [comment]
Just to clarify, this issue is not advocating to deprecate the `ws://` endpoints, this should still be there since it's used for relayers for example
